### PR TITLE
Radiancemeter sensor plugin

### PR DESCRIPTION
**Pull request by leroyvn**
This commit ports Mitsuba 1's radiance meter sensor.

Co-authored-by: @schunkes

### CODE REVIEW

File: src/sensors/radiancemeter.cpp [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+        // 2. Set ray origin and direction
+        auto trafo = m_world_transform->eval(time, active);
+        ray.o      = trafo.transform_affine(Point3f{ 0.f, 0.f, 0.f });
+        ray.d      = trafo.transform_affine(Vector3f{ 0.f, 0.f, 1.f });
+
+        // 3. Set differentials
+        ray.d_x               = 0.f;
+        ray.d_y               = 0.f;
+        ray.has_differentials = true;
```
Comment (Speierers) [inline src/sensors/radiancemeter.cpp:125]: Not sure whether we should even set the derivatives here. 
As this sensor operates on a single pixel it makes sense to that `d_x, d_y` should be equal to zero.

Maybe we could simply leave those with their default values (`ray.has_differentials` will be `false`)?

**Comment by Speierers** [comment]
Hi @leroyvn, 

Thanks a lot for opening this PR. Really happy to see people porting/implementing more plugins in Mitsuba 2!

The code looks great to me, and the tests as well. Just one minor comment (see above).

I am curious to know if you have encountered any bugs when rendering 1x1 images with this plugin. Not sure this is something we have ever tested.

**Comment by leroyvn** [comment]
I had similar behaviour as in #82 with both the `perspective` and `radiancemeter` plugins, but I don't think it's due to the film size. 

Forcing the use of a `box` reconstruction filter could be good. Maybe I should change something to implement this behaviour?

**Comment by Speierers** [comment]
> Forcing the use of a box reconstruction filter could be good. Maybe I should change something to implement this behaviour?

Indeed it would be nice to have a warning in the constructor if the `film->reconstruction_filter()` isn't a box filter. Or maybe it is enough to check whether the `radius` of that filter is `<= 0.5f`?

One other thing: could we add a test where we construct a simple scence (e.g. no shapes, just a `constant` emitter) with this sensor, and call `scene.integrator().render()`?

**Comment by leroyvn** [comment]
Sure, I will do do that.

**Comment by leroyvn** [comment]
There it is! I used numpy to check the results, is that okay?

File: src/sensors/radiancemeter.cpp [hunk 2]
```diff
@@ -77,6 +77,11 @@ MTS_VARIANT class RadianceMeter final : public Sensor<Float, Spectrum> {
 
         if (m_film->size() != ScalarPoint2i(1, 1))
             Throw("This sensor only supports films of size 1x1 Pixels!");
+
+        if (m_film->reconstruction_filter()->radius() >
+            0.5f + math::RayEpsilon<Float>)
```
Comment (Speierers) [inline src/sensors/radiancemeter.cpp:82]: Not sure the `math::RayEpsilon` is necessary here. Am I missing someting?

I would just do the following comparison:
```
if (m_film->reconstruction_filter()->radius() > 0.5f)
```

**Comment by Speierers** [comment]
> There it is! I used numpy to check the results, is that okay?

Yes, `numpy` is fine when dealing with `Bitmap`.

One minor comment regarding the filter radius check (see above) and it would be great to squash those commit into one afterthat.

Then this should be ready to launch! 🚀 

Thanks again @leroyvn

File: src/sensors/radiancemeter.cpp (hunk 2 above)
Comment (leroyvn) [reply src/sensors/radiancemeter.cpp:82]: I initially didn't put the extra `math::RayEpsilon`, but then I was getting the warning when using the box filter with its default settings; hence this fix.

Comment (Speierers) [reply src/sensors/radiancemeter.cpp:82]: My bad, I didn't know `math::RayEpsilon` was used in `box.cpp`. All good then!

**Pull request merged**